src/cpp/client.cpp

```diff
@@ -590,6 +706,36 @@ void Client::run_model(const std::string& name,
     _redis_server->run_model(key, inputs, outputs);
 }
 
+// Run a model in the database using the
+// specified input and output tensors in a multi-GPU system
+void Client::run_model_multigpu(const std::string& name,
+                                std::vector<std::string> inputs,
+                                std::vector<std::string> outputs,
+                                int image_id,
+                                int first_gpu,
+                                int num_gpus)
+{
+    if (first_gpu < 0) {
+        throw SRParameterException("first_gpu must be a non-negative integer.");
+    }
+    if (num_gpus < 1) {
+        throw SRParameterException("num_gpus must be a positive integer.");
+    }
+
+    if (image_id < 1) {
```

Comment:
Because you're relying on modular arithmetic, negative numbers should be supportable. MPI ranks I believe start with 0, so this could raise an unintended error. Probably just delete this check

Reply:
done